### PR TITLE
Improved test cases by using paramerters

### PR DESCRIPTION
**Pull request by AdamLouly**
### Description
Completing some missing parts of some test cases for python bindings

### Motivation and Context
Some test cases like test_training_module_checkpoint and test_optimizer step were not completed before because we had no access to parameters to check if the parameters are changing after the optimizer step or that the checkpoint saved parameters remains the same.
now that we have access to the vector or parameters by exposing get_contiguous_parameters() method.
we can complete the tests.

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_python_bindings.py [hunk 1]
```diff
@@ -137,9 +137,13 @@ def test_optimizer_step():
         optimizer = Optimizer(optimizer_file_path, model)
 
         model.train()
+        old_params = model.get_contiguous_parameters()
```
Comment (pengwa) [inline orttraining/orttraining/test/python/orttraining_test_python_bindings.py:140]: old_flatten_params?

File: orttraining/orttraining/test/python/orttraining_test_python_bindings.py [hunk 2]
```diff
@@ -226,10 +230,19 @@ def test_training_module_checkpoint():
         checkpoint_save_path = os.path.join(temp_dir, "checkpoint_export.ckpt")
 
         model.save_checkpoint(checkpoint_save_path)
+        old_params = model.get_contiguous_parameters()
 
-        # TODO : Load checkpoint to a zeroed model and assert parameters are different.
+        # Assert the checkpoint was saved.
         assert os.path.exists(checkpoint_save_path)
 
+        # Assert the checkpoint parameters remains after saving.
```
Comment (pengwa) [inline orttraining/orttraining/test/python/orttraining_test_python_bindings.py:238]: --> remain

File: orttraining/orttraining/test/python/orttraining_test_python_bindings.py [hunk 3]
```diff
@@ -226,10 +230,19 @@ def test_training_module_checkpoint():
         checkpoint_save_path = os.path.join(temp_dir, "checkpoint_export.ckpt")
 
         model.save_checkpoint(checkpoint_save_path)
+        old_params = model.get_contiguous_parameters()
 
-        # TODO : Load checkpoint to a zeroed model and assert parameters are different.
+        # Assert the checkpoint was saved.
```
Comment (pengwa) [inline orttraining/orttraining/test/python/orttraining_test_python_bindings.py:235]: looks this change did complete the TODO "TODO : Load checkpoint to a zeroed model and assert parameters are different."

Comment (AdamLouly) [reply orttraining/orttraining/test/python/orttraining_test_python_bindings.py:235]: the change I did will test the same thing, there's no need to create a zeroed model and load it to test if the checkpoint params remain the same.

**Review by pengwa — APPROVED** [review]

**Comment by AdamLouly** [comment]
Thanks for the review

**Pull request merged**